courier/template/load_template.go

```diff
@@ -65,6 +85,32 @@ func loadBuiltInTemplate(filesytem fs.FS, name string, html bool) (Template, err
 	return tpl, nil
 }
 
+func loadRemoteTemplate(url string, name string, html bool, root string) (Template, error) {
+	if t, found := cache.Get(name); found {
+		return t.(Template), nil
+	}
+
+	f := fetcher.NewFetcher()
```

Comment:
This is vulnerable to SSRF, please use the global HTTP client from the registry :) Here's an example:

```
fetch := fetcher.NewFetcher(fetcher.WithClient(s.d.HTTPClient(r.Context())))
```

Reply:
So looking into this a bit. When using the `registryDefault.HTTPClient(ctx)` it returns a `*retryablehttp.Client` which is not supported by fetcher. I would have to add an optional function in ory/x for this

```go
func WithResilientClient(rc *retryablehttp.Client) func(*opts) {
	return func(o *opts) {
		o.hc = rc
	}
}
```

Does that make sense? Also, it seems doing so would break a lot of things in Kratos as there are many changes to ory/x

Reply:
When just returning an `HTTPClient` from the `retryablehttp.Client` it removes a lot of the setup such as max retries etc.

```go
f := fetcher.NewFetcher(fetcher.WithResilientClient(d.HTTPClient(ctx).HTTPClient))
```

Reply:
based on branch v0.8 solved this